pkg/apis/experimental/validation/validation.go

```diff
@@ -333,3 +335,50 @@ func ValidateJobSpecUpdate(oldSpec, spec experimental.JobSpec) errs.ValidationEr
 	}
 	return allErrs
 }
+
+func ValidateDerivedNodeMetricsName(name string, prefix bool) (bool, string) {
+	return apivalidation.ValidateNodeName(name, prefix)
+}
+
+func validateMetricsWindows(mw experimental.MetricsWindows, field string) errs.ValidationErrorList {
+	result := errs.ValidationErrorList{}
+	if mw.EndTime.Equal(unversioned.Time{}) {
+		result = append(result, errs.NewFieldInvalid(field+".endTime", mw.EndTime, `is not set`))
+	}
+	for _, metrics := range mw.Windows {
+		fields := map[string]experimental.ResourceUsage{
+			"mean": metrics.Mean,
+			"max":  metrics.Max,
+			"95th": metrics.NinetyFive,
+		}
+		for f, resUsage := range fields {
+			resources := map[string]bool{
+				"cpu":    false,
+				"memory": false,
+			}
+			for n := range resUsage {
+				resources[n] = true
+			}
+			for n, b := range resources {
+				if !b {
+					result = append(result, errs.NewFieldInvalid(field+"."+f, resUsage, fmt.Sprintf(`is missing '%s' resource`, n)))
+				}
+			}
+		}
+	}
+	return result
+}
+
+func ValidateDerivedNodeMetrics(metrics *experimental.DerivedNodeMetrics) errs.ValidationErrorList {
+	result := errs.ValidationErrorList{}
```

Comment:
Having it there if it's only optional seems wrong to me. The idea is that since the windows can be out of date by up to more than a minute, not knowing how out of date they are can be a problem.